src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -671,6 +677,33 @@ void mutate(const gsl::not_null<DataBox<TagList>*> box, Invokable&& invokable,
                                               tmpl::pin<full_mutated_items>,
                                               tmpl::get_source<tmpl::_1>>>,
                       tmpl::get_destination<tmpl::_1>>>;
+  if constexpr (not std::is_same_v<
+                    decltype(invokable(
+                        make_not_null(
+                            &box->template get_item<detail::first_matching_tag<
+                                 TagList, MutateTags>>()
+                                 .mutate())...,
+                        std::forward<Args>(args)...)),
+                    void>) {
+    decltype(auto) return_value = invokable(
+        make_not_null(&box->template get_item<
+                              detail::first_matching_tag<TagList, MutateTags>>()
+                           .mutate())...,
+        std::forward<Args>(args)...);
+
+    EXPAND_PACK_LEFT_TO_RIGHT(box->template mutate_mutable_subitems<MutateTags>(
+        typename Subitems<MutateTags>::type{}));
+    box->template reset_compute_items_after_mutate(
+        first_compute_items_to_reset{});
+
+    box->mutate_locked_box_ = false;
+    return return_value;
+  }
```

Comment:
should the rest go in an `else` branch of the `if constexpr` so it's not compiled in those cases?